core/src/View/Kokkos_BasicView.hpp

```diff
@@ -0,0 +1,611 @@
+// However they are not part of the public standard interface
+template <class T>
+struct is_layout_right_padded : public std::false_type {};
+
+template <size_t Pad>
+struct is_layout_right_padded<Kokkos::Experimental::layout_right_padded<Pad>>
+    : public std::true_type {};
+
+template <class T>
+struct is_layout_left_padded : public std::false_type {};
+
+template <size_t Pad>
+struct is_layout_left_padded<Kokkos::Experimental::layout_left_padded<Pad>>
+    : public std::true_type {};
+
+}  // namespace Impl
+
+template <class ElementType, class Extents, class LayoutPolicy,
+          class AccessorPolicy>
+class BasicView {
+ public:
+  using mdspan_type =
+      mdspan<ElementType, Extents, LayoutPolicy, AccessorPolicy>;
+  using extents_type     = typename mdspan_type::extents_type;
+  using layout_type      = typename mdspan_type::layout_type;
+  using accessor_type    = typename mdspan_type::accessor_type;
+  using mapping_type     = typename mdspan_type::mapping_type;
+  using element_type     = typename mdspan_type::element_type;
+  using value_type       = typename mdspan_type::value_type;
+  using index_type       = typename mdspan_type::index_type;
+  using size_type        = typename mdspan_type::size_type;
+  using rank_type        = typename mdspan_type::rank_type;
+  using data_handle_type = typename mdspan_type::data_handle_type;
+  using reference        = typename mdspan_type::reference;
+  using memory_space     = typename accessor_type::memory_space;
+  using execution_space  = typename memory_space::execution_space;
+
+  KOKKOS_FUNCTION static constexpr rank_type rank() noexcept {
+    return extents_type::rank();
+  }
+  KOKKOS_FUNCTION static constexpr rank_type rank_dynamic() noexcept {
+    return extents_type::rank_dynamic();
+  }
+  KOKKOS_FUNCTION static constexpr size_t static_extent(rank_type r) noexcept {
+    return extents_type::static_extent(r);
+  }
+  KOKKOS_FUNCTION constexpr index_type extent(rank_type r) const noexcept {
+    return map.extents().extent(r);
+  };
+
+ protected:
+  // These are pre-condition checks which are in Kokkos::View 4.4 unconditional
+  // - i.e. in release mode
+  template <class OtherMapping>
+  KOKKOS_FUNCTION static constexpr void check_basic_view_constructibility(
+      const OtherMapping &rhs) {
+    using src_t          = typename OtherMapping::layout_type;
+    using dst_t          = layout_type;
+    constexpr size_t rnk = mdspan_type::rank();
+    if constexpr (!std::is_same_v<src_t, dst_t>) {
+      if constexpr (Impl::is_layout_left_padded<dst_t>::value) {
+        if constexpr (std::is_same_v<src_t, layout_stride>) {
+          index_type stride = 1;
+          for (size_t r = 0; r < rnk; r++) {
+            if (rhs.stride(r) != stride)
+              Kokkos::abort("View assignment must have compatible layouts");
+            if constexpr (rnk > 1)
+              stride *= (r == 0 ? rhs.stride(1) : rhs.extents().extent(r));
+          }
+        }
```

Comment:
No else block? We only check if the source is LayoutStride? What if it's `LayoutRight`?

Reply:
we only allow LayoutRight to LayoutLeft for Rank-1 in which case its trivial. The only other situation you could do transforms is if you have degenerated Views anyway (i.e. 1x1xNx1) or so.